documentation/Docker-DNS.md

```diff
@@ -0,0 +1,86 @@
+# Docker DNS setup
+
+There are several DNS components in the Docker setup:
+
+* A non-validating resolver, used for DNS resolving by almost all tests.
+  As we have our own DNSSEC validation test, we want to see bogus responses as well.
+* A validating resolver, used to validate DANE records through ldns-dane.
+* An authoritative server for the connection test zone.
+* The DNS records in the zone for `INTERNETNL_DOMAINNAME`.
+  These are hosted elsewhere, but have certain requirements for the instance to work.
+
+The resolvers do not require any specific configuration.
+In this document, `INTERNETNL_DOMAINNAME` is `example.com`.
+
+## Common parts
+
+For accessing the absolute minimum basic functionality of the application the following DNS records must be configured:
+
+    example.com.                    A      192.0.2.1
+                                   AAAA   2001:db8:1::1
+
+The following extra records must be configured for language switching:
+
+    www.example.com.                CNAME  example.com.
+    nl.example.com.                 CNAME  example.com.
+    en.example.com.                 CNAME  example.com.
+
+The hostname (`INTERNETNL_DOMAINNAME`) should have SPF, DKIM and DMARC,
+as some mail servers may filter on this, and it could affect mail tests.
+For a domain that does not otherwise send email, use:
+
+    example.com.		           TXT	"v=spf1 a -all"	; The "a" mechanism is needed for the mail test (see rfc7208, section-2.3).
+    _domainkey.example.com.	       TXT	"v=DKIM1; p="	; empty DKIM to score 100% for this non-sending subdomain that does have SPF "a" mechanism which is needed for mail test.
```

Comment:
maybe better to use *._domainkey? (see https://www.cloudflare.com/learning/dns/dns-records/protect-domains-without-email/)

Reply:
Yes, although not necessary, we could do so. I will also change this in our own DNS (https://github.com/internetstandards/internet.nl-DNS-zone-file/issues/85).

Note that it is more or less a vanity record to score 100% for the instance-domain in the mailtest itself. (Note that the instance-domain does need the "a" in the SPF record for the working of the mailtest.  Therefore, the Internet.nl mailtest does not see the instance-domain as a non-sending domain and requires a DKIM record.)

Furthermore, note that M3AAWG previously promoted *._domainkey for parked domains. However, they have changed their advice into not publishing a DKIM record at all for parked domains. See:
 
- 2015: https://www.m3aawg.org/sites/default/files/m3aawg_parked_domains_bp-2015-12.pdf
- 2022: https://www.m3aawg.org/sites/default/files/m3aawg_parked_domains_bcp-2022-06.pdf